policies/versioning-policy.md

```diff
@@ -0,0 +1,133 @@
+project from version 3.0.0 onwards.  It also details the level of [ABI] and
+[API] compatibility each version represents.
+
+> _Note: All examples herein are illustrative and do not constitute part of the
+> versioning policy._
+
+The version scheme consists a triple of numbers:
+**[major]**.**[minor]**.**[patch]**.
+
+> For example, the version _3.0.1_ has a _major_ version of _3_, a _minor_
+> version of _0_ and a _patch_ version of _1_.
+
+This closely aligns with the expectations of users who are familiar
+with _semantic versioning_.  However, it is not quite _semantic versioning_
+because adopting _semantic versioning_ would mean changing our current
+Long Term Support ([LTS]) policies and practices.
+
+## Major release
+
+A major release is indicated by changing the _first digit_ of the version.
+A major release can, and generally will, introduce API and ABI breaking
+changes.
+
+> For example, a program that runs with OpenSSL 3.0.1 should not expect to run
+> with OpenSSL 4.0.0 without modifications.  If it does run, it might not be
+> as efficient as it was.
+
+In order to maintain stability and limit rework across major versions:
+
+- Existing [public interfaces] will remain unmodified except where changes
+  are unlikely to break source compatibility or where structures are being
+  made opaque.
+- No existing public interface can be removed until its replacement has
+  been in place in an LTS stable release. The original interface must
+  also have been documented as deprecated for at least 5 years.
+- When structures are made opaque, any newly required accessor macros
+  or functions are added in a feature release of the extant LTS release
+  and all supported intermediate successor releases.
+
+Exceptions to these rules require a vote by the [OMC].  See also the
+[stable release update policy].
+
+## Minor release
+
+A minor release is indicated by changing the _second_ digit of the version.
+A minor release can, and generally will, introduce new features.  However both
+the API and ABI will be preserved.
+
+> For example, a program built with OpenSSL release 3.0.1 will be able to
+> run with OpenSSL 3.1.0 but might not be able to take advantage of new
+> features without modification.
+
+Exceptions to these rules require a vote by the OMC.
+
+## Patch release
+
+A patch release is indicated by changing the _final_ digit of the version.
+A patch release will only contain bug and security fixes.
+Both the API and ABI will remain compatible across patch releases.
+
+> For example, a program linked with OpenSSL release 3.0.0 can
+> run with OpenSSL 3.0.1 without changes.
+
+Exceptions to these rules require a vote by the OMC.
+
+## Long term stable release
+
+A release can be designated as a release.  LTS releases will be supported
+for at least five years and the project will specify a LTS release at
+least every four years.  Non-LTS releases will be supported for at least
+two years.
```

Comment:
The title of this section is "Long term stable releases". But it actually talks about the support period for non-LTS releases too. Also the para below is applicable to both LTS and non-LTS releases.

Reply:
I'll put a "supported releases" section in, although the flow will still be a bit odd.